### PR TITLE
MongoDB as MD backend support

### PR DESCRIPTION
**Pull request by vrancurel**
- Add official Mongo image (stable)
- Add MD Mongo Scality image that supports level-net-rpc

### CODE REVIEW

**Comment by vrancurel** [comment]
Initial Proposal.

There are still minor integration related issues to discuss in all those PRs:

- The scality MD image: https://github.com/scality/mdmongo/pull/1
- The fork of mongodown w/ modifications: https://github.com/watson/mongodown/pull/9 

Especially I would like to draw the attention on the way we are storing object values (currently as a string) which poses some troubles and especially the need to explode JSON values in mongodown. The effect of that is that mongodown tests break

@rahulreddy @ploki   any ideas ?

File: images/metadata/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "devDependencies": {
+    "eslint": "^2.4.0",
+    "eslint-config-airbnb": "^6.0.0",
+    "eslint-config-scality": "scality/Guidelines",
+    "eslint-plugin-react": "4.2.3"
+  },
+  "dependencies": {
+    "arsenal": "scality/Arsenal",
+    "levelup": "^1.3.9",
+    "mongodown": "scality/mongodown#ft/makeItWorking",
```
Comment (jonathan-gramain) [inline images/metadata/package.json]: remove the branch tag? Or make it part of a dropme commit

**Comment by vrancurel** [comment]
Now supports versioning.
One issue though is that there is no method available for deleting a collection ( @jonathan-gramain ?)

**Comment by vrancurel** [comment]
Mongo will be implemented as an S3 metadata backend

**Pull request closed without merging**